src/Raven.Client/Documents/Indexes/IndexFailureInformation.cs

```diff
@@ -23,7 +23,7 @@ public class IndexFailureInformation
         /// Indicates whether this is invalid index.
         /// </summary>
         /// <value><c>true</c> if this is invalid index; otherwise, <c>false</c>.</value>
-        public bool IsInvalidIndex(Func<bool> isStale)
```

Comment:
We shouldn't change this. It was not added without a reason. Probably we are substituting one issue with another.

Reply:
I want to avoid the indirection here, and I checked, we preserve the same behavior

Reply:
No, you are now calculating staleness everytime instead of doing it at later stage of that function.

Reply:
It doesn't actually matter, this is only called on index error. 
I don't think we need to worry about perf here.

Which means that the indirection argument is also invalid.

Need to consider which option is clearer to maintain.

This way, we don't need to worry about the state of the func that we pass, which we previous dealt with.